examples/demo-rollup/src/rollup.rs

```diff
@@ -22,10 +26,21 @@ use crate::register_rpc::register_ethereum;
 use crate::register_rpc::{register_ledger, register_sequencer};
 use crate::{get_genesis_config, initialize_ledger, ROLLUP_NAMESPACE};
 
+type AppVerifier<DA, Vm> = StateTransitionVerifier<
+    AppTemplate<
+        ZkDefaultContext,
+        <DA as DaService>::Spec,
+        <Vm as ProofSystem>::Guest,
+        Runtime<ZkDefaultContext>,
+    >,
+    <DA as DaService>::Verifier,
+    Vm,
+>;
+
 /// Dependencies needed to run the rollup.
-pub struct Rollup<Vm: Zkvm, DA: DaService + Clone> {
+pub struct Rollup<Vm: ProofSystem, DA: DaService + Clone> {
     /// Implementation of the STF.
-    pub app: App<Vm, DA::Spec>,
+    pub app: App<Vm::Host, DA::Spec>,
     /// Data availability service.
```

Comment:
Just for uniformity, should we put `Option<(<Vm as ProofSystem>::Host, AppVerifier<DA, Vm>)>` as a field in Rollup? This way, `App` & `AppVerifier` are goe together.